internal/scm/azuredevops/azuredevops.go

```diff
@@ -0,0 +1,540 @@
+		lCopy := label
+		labels[i] = core.WebApiTagDefinition{
+			Active: &activeLabel,
+			Name:   &lCopy,
+		}
+	}
+
+	srn := PrependPrefixIfNeeded(updatedPR.Head) // Pass the value of newPR.Head
+	trn := PrependPrefixIfNeeded(updatedPR.Base)
+
+	updateResult, err := a.gitClient.UpdatePullRequest(ctx, git.UpdatePullRequestArgs{
+		GitPullRequestToUpdate: &git.GitPullRequest{
+			Title:         &prTitle,
+			Description:   &updatedPR.Body,
+			SourceRefName: &srn,
+			TargetRefName: &trn,
+			Labels:        &labels,
+			Reviewers:     &reviewers,
+		},
+		RepositoryId:  &pr.repoID,
+		PullRequestId: &pr.id,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pullRequest{
+		ownerName:  r.ownerName,
+		repoName:   r.name,
+		repoID:     r.rid,
+		branchName: updatedPR.Head,
+		id:         *updateResult.PullRequestId,
+	}, nil
+}
+
+func (a *AzureDevOps) getUserIds(ctx context.Context, usernames []string) ([]string, error) {
+	userIds := make([]string, len(usernames))
+
+	searchFilter := "General"
+
+	for i, username := range usernames {
+		uCopy := username
+		users, err := a.identityClient.ReadIdentities(ctx, identity.ReadIdentitiesArgs{
+			SearchFilter: &searchFilter,
+			FilterValue:  &uCopy,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+		if len(*users) != 1 {
+			return nil, fmt.Errorf("could not find user: %s", usernames[i])
+		}
+
+		userIds[i] = (*users)[0].Id.String()
+	}
+	return userIds, nil
+}
+
+func (a *AzureDevOps) GetPullRequests(ctx context.Context, branchName string) ([]scm.PullRequest, error) {
+	repos, err := a.getRepos(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	bn := PrependPrefixIfNeeded(branchName)
+
+	prs := []scm.PullRequest{}
+	for _, repo := range *repos {
+		pr, err := a.getPullRequest(ctx, bn, repo.Id.String())
+		if err != nil {
+			return nil, err
+		}
+		if pr == nil {
+			continue
+		}
+
+		prs = append(prs, convertPullRequest(pr))
+	}
+
+	return prs, nil
+}
+
+func convertPullRequest(pr *git.GitPullRequest) pullRequest {
+	return pullRequest{
+		ownerName:               *pr.Repository.Project.Name,
+		repoName:                *pr.Repository.Name,
+		repoID:                  pr.Repository.Id.String(),
+		branchName:              *pr.SourceRefName,
+		id:                      *pr.PullRequestId,
+		status:                  pullRequestStatus(pr),
+		lastMergeSourceCommitID: *pr.LastMergeSourceCommit.CommitId,
+	}
+}
+
+func (a *AzureDevOps) getPullRequest(ctx context.Context, branchName string, rid string) (*git.GitPullRequest, error) {
+	status := &git.PullRequestStatusValues.Active
+
+	prs, err := a.gitClient.GetPullRequests(ctx, git.GetPullRequestsArgs{
+		RepositoryId: &rid,
+		SearchCriteria: &git.GitPullRequestSearchCriteria{
+			SourceRefName: &branchName,
+			Status:        status,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if len(*prs) == 0 {
+		return nil, nil
+	}
+
+	return &(*prs)[0], nil
+}
+
+func pullRequestStatus(pr *git.GitPullRequest) scm.PullRequestStatus {
```

Comment:
Same as for `convertPullRequest`
```suggestion
func pullRequestStatus(pr git.GitPullRequest) scm.PullRequestStatus {
```